ocpp_codec/validators.py

```diff
@@ -216,3 +216,21 @@ def to_json(self, value):
                 value=value, enum_class=self.enum_class.__name__,
             )
         return value.value
+
+
+class OutgoingMessageDecimalEncoder(BaseEncoder):
+    """Encoder to limit the precision of decimal values sent to charge points.
+
+    This encoder doesn't restricts the precision of float values received from the charge point, but limits it to 6
+    places when sending a value, as required by OCPP 2.0 spec (section 2.1.3).
+    """
+
+    def from_json(self, json_value):
+        return json_value
+
+    def to_json(self, value):
+        if not isinstance(value, float):
```

Comment:
Maybe use a `try/except` around a `float(value)` so we can give any object that can pass for a float instead?

Reply:
Agreed

Reply:
On second thought, the value's type is already checked before calling the encoder (here: https://github.com/Polyconseil/ocpp-codec/blob/master/ocpp_codec/serializer.py#L116-L120). I think I'll drop the check entirely

Reply:
Whoops I mixed things up. There should be no validation in `from_json`, but there should be in `to_json`, forgot about that. Your idead is good, I'll fix this